docs/gallery/general/plot_read_demo.py

```diff
@@ -0,0 +1,261 @@
+# relative to the session start time.
+
+print(units["spike_times"][0])
+
+####################
+# Visualize spiking activity relative to stimulus onset
+# --------
+# We can look at when these single units spike relative to when image stimuli were presented to the subject.
+# We will iterate over the first 3 units and get their spike times.
+# Then for each unit, we iterate over each stimulus onset time and compute the spike times relative
+# to stimulus onset. Finally, create a raster plot and histogram of these aligned spike times.
+
+before = 1.  # in seconds
+after = 3.
+
+# Get the stimulus times for all stimuli
+stim_on_times = stimulus_presentation.timestamps[:]
+
+for unit in range(3):
+    unit_spike_times = nwbfile.units["spike_times"][unit]
+    trial_spikes = []
+    for time in stim_on_times:
+        # Compute spike times relative to stimulus onset
+        aligned_spikes = unit_spike_times - time
+        # Keep only spike times in a given time window around the stimulus onset
+        aligned_spikes = aligned_spikes[
+            (-before < aligned_spikes) & (aligned_spikes < after)]
+        trial_spikes.append(aligned_spikes)
+    fig, axs = plt.subplots(2, 1, sharex="all")
+    plt.xlabel('time (s)')
+    axs[0].eventplot(trial_spikes)
+
+    axs[0].set_ylabel('trial')
+    axs[0].set_title('unit {}'.format(unit))
+    axs[0].axvline(0, color=[.5, .5, .5])
+
+    axs[1].hist(np.hstack(trial_spikes), 30)
+    axs[1].axvline(0, color=[.5, .5, .5])
+
+####################
+# Access Trials
+# --------
+# Trials are stored as :py:class:`~pynwb.epoch.TimeIntervals` object which is a subclass
+# of :py:class:`~hdmf.common.table.DynamicTable`. :py:class:`~hdmf.common.table.DynamicTable` objects are used to store
+# metadata about each trial in a tabular form, where each row represents a trial and has a start time, stop time,
+# and additional metadata.
+#
+# .. seealso::
+#     You can learn more about trials in the :ref:`basic_trials` tutorial section.
+#
+# Similarly to :py:class:`~pynwb.misc.Units` we can view trials as a :py:class:`pandas.DataFrame`.
+
+trials_df = nwbfile.trials.to_dataframe()
+print(trials_df)
+
+####################
+# The :py:class:`~pynwb.file.NWBFile.stimulus` can be mapped one to one to each row (trial)
+# of :py:class:`~pynwb.file.NWBFile.trials` based on the ``stim_on_time`` column.
+
+assert np.all(stimulus_presentation.timestamps[:] == trials_df.stim_on_time[:])
+
+####################
+# Visualize the first 3 images that were categorized as landscapes in the session:
+
+stim_on_times_landscapes = trials_df[trials_df.category_name == 'landscapes'].stim_on_time
+for time in stim_on_times_landscapes[:3]:
+    img = np.squeeze(stimulus_presentation.data[np.where(stimulus_presentation.timestamps[:] == time)])
+    # Reverse the last dimension because the data were stored in BGR instead of RGB
+    img = img[..., ::-1]
+    plt.figure()
+    plt.imshow(img, aspect='auto')
+
+####################
```

Comment:
We should add a link to https://nwb-overview.readthedocs.io/en/latest/tools/tools_home.html for details somewhere in the intro of this section for details.